cognite/seismic/protos/v1/seismic_service_messages.proto

```diff
@@ -167,9 +167,27 @@ message VolumeRequest {
         int64 seismic_store_id = 2;  // Need to be a data manager or tenant user, not 3rd party, to access by tracestore
     }
     LineBasedVolume volume = 3;
+    GeometryFilter geometry_filter = 5; // Mutually exclusive with volume
     bool include_trace_header = 4;
 }
 
+message GeometryFilter {
+    oneof filter {
+        LineSlice slice = 1;
+        VolumeCube cube = 2;
+    }
+    LineDescriptor z_range = 3;
+}
+
+message LineSlice {
+    Geometry line = 1;
```

Comment:
What happens here if a non line geometry gets passed in? I'm guessing it returns an error? Might be worth adding a comment to mention for the documentation

Reply:
Yeah, I guess that's the idea.